### PR TITLE
Add pullapprove

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR adds pullapprove to this repo. It is useful for automatic review requests.

Testing steps:
- testing can be done in the commands below

Acceptance Criteria:
- first criteria (maybe a test case)
- second criteria
- third citeria

Solves issue ~~#1~~ there is no existing issue for this.

Steps needed for migration:
- no mirgation needed

### CODE REVIEW

**Comment by MichaelBuessemeyer** [comment]
/pullapprove-test

```
version: 3

extends: https://api.github.com/repos/example/pullapprove-config/contents/.pullapprove.yml

- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this asap.

groups:
  S⊙w⊙T:
    conditions:
    - "'S⊙w⊙T' in labels"
    reviewers:
      users:
      - grittaweisheit
      - Xantocx
      - JoeAtHPI
      - lpfennigschmidt
      - MichaelBuessemeyer
    reviews:
      required: 1
      request: 1  
      request_order: random 
  PO-AD:
    conditions:
    - "'S⊙w⊙T' in groups.approved"
    reviewers:
      users:
      - ADimeo
      teams:
      - team-S⊙w⊙T
    reviews:
      required: 1
      request: 1
      request_order: given  
```

**Comment by WorldofJARcraft** [comment]
/pullapprove-test
version: 3

- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this asap.

groups:
  S⊙w⊙T:
    conditions:
    - "'S⊙w⊙T' in labels"
    reviewers:
      users:
      - grittaweisheit
      - Xantocx
      - JoeAtHPI
      - lpfennigschmidt
      - MichaelBuessemeyer
    reviews:
      required: 1
      request: 1  
      request_order: random 
  PO-AD:
    conditions:
    - "'S⊙w⊙T' in groups.approved"
    reviewers:
      users:
      - ADimeo
      teams:
      - team-S⊙w⊙T
    reviews:
      required: 1
      request: 1
      request_order: given

**Comment by WorldofJARcraft** [comment]
/pullapprove-test
version: 3

when: pullapprove.approved
comment: "The review is completed. @{{ author }}, please merge this asap.
groups:
S⊙w⊙T:
conditions:
- "'S⊙w⊙T' in labels"
reviewers:
users:
- grittaweisheit
- Xantocx
- JoeAtHPI
- lpfennigschmidt
- MichaelBuessemeyer
reviews:
required: 1
request: 1
request_order: random
PO-AD:
conditions:
- "'S⊙w⊙T' in groups.approved"
reviewers:
users:
- ADimeo
teams:
- team-S⊙w⊙T
reviews:
required: 1
request: 1
request_order: given

**Comment by WorldofJARcraft** [comment]
/pullapprove-test

```
version: 3

- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this asap.

groups:
  SoWoT:
    conditions:
    - "'SoWoT' in labels"
    reviewers:
      users:
      - grittaweisheit
      - Xantocx
      - JoeAtHPI
      - lpfennigschmidt
      - MichaelBuessemeyer
    reviews:
      required: 1
      request: 1  
      request_order: random 
  PO-AD:
    conditions:
    - "'SoWoT' in groups.approved"
    reviewers:
      users:
      - ADimeo
      teams:
      - team-SoWoT
    reviews:
      required: 1
      request: 1
      request_order: given  
```

File: .pullapprove.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+version: 3
+
+extends: https://api.github.com/repos/example/pullapprove-config/contents/.pullapprove.yml
+
+- when: pullapprove.approved
+  comment: "The review is completed. @{{ author }}, please merge this asap.
```
Comment (adrianjost) [inline .pullapprove.yml]:
```suggestion
  comment: "The review is completed. @{{ author }}, please merge this asap."
```

**Comment by adrianjost** [comment]
/pullapprove-test
```
version: 3

- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this asap."

groups:
  S⊙w⊙T:
    conditions:
    - "'S⊙w⊙T' in labels"
    reviewers:
      users:
      - grittaweisheit
      - Xantocx
      - JoeAtHPI
      - lpfennigschmidt
      - MichaelBuessemeyer
    reviews:
      required: 1
      request: 1  
      request_order: random 
  PO-AD:
    conditions:
    - "'S⊙w⊙T' in groups.approved"
    reviewers:
      users:
      - ADimeo
      teams:
      - team-S⊙w⊙T
    reviews:
      required: 1
      request: 1
      request_order: given
```

**Comment by adrianjost** [comment]
/pullapprove-test
```
version: 3

notifications:
- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this asap."

groups:
  S⊙w⊙T:
    conditions:
    - "'S⊙w⊙T' in labels"
    reviewers:
      users:
      - grittaweisheit
      - Xantocx
      - JoeAtHPI
      - lpfennigschmidt
      - MichaelBuessemeyer
    reviews:
      required: 1
      request: 1  
      request_order: random 
  PO-AD:
    conditions:
    - "'S⊙w⊙T' in groups.approved"
    reviewers:
      users:
      - ADimeo
    reviews:
      required: 1
      request: 1
      request_order: given
```

**Comment by adrianjost** [comment]
/pullapprove-test
```
version: 3

pullapprove_conditions:
- condition: "'WIP' not in labels"
  unmet_status: pending
  explanation: "Work in progress"
- condition: "'- [ ]' not in body"
  unmet_status: failure
  explanation: "Please finish all the required checklist tasks"
#- condition: "'Travis*' in statuses.succeeded" # Not working yet
#  unmet_status: failure
#  explanation: "Tests must pass before review starts"
#- condition: "'stickler*' in statuses.succeeded"
#  unmet_status: failure
#  explanation: "Linter must pass before review starts"


notifications:
- when: pull_request.opened
  comment: |
    Hey @{{ author }}, thanks for the PR! The review will start once
    the tests, CI checks and PR requirements (see checklist in your PR) have passed.
- when: pullapprove.approved
  if: "author_association == 'CONTRIBUTOR'"
  comment: "The review is completed. Thanks @{{ author }}, we'll take it from here."

groups:
  client-core:
    reviewers:
      teams:
      - client-core
    reviews:
      required: 1  # number of approvals required from this group
      request: 1  # number of review requests sent at a time
      request_order: shuffle

  ux-core:
    conditions:
    - "'**/*.hbs' in files or 'static/styles/**/*.scss' in files or '.pullapprove.yml' in files"
    # only review if "**/*.hbs", "static/styles/**/*.scss" files or ".pullapprove.yml" have been touched
    reviewers:
      teams:
      - ux-core
    reviews:
      required: 1  # number of approvals required from this group
      request: -1  # request from all members a review
```

**Comment by adrianjost** [comment]
/pullapprove-test
```
version: 3

pullapprove_conditions:
- condition: "'WIP' not in labels"
  unmet_status: pending
  explanation: "Work in progress"


notifications:
- when: pull_request.opened
  comment: |
    Hey @{{ author }}, thanks for the PR! The review will start once
    the tests, CI checks and PR requirements (see checklist in your PR) have passed.
- when: pullapprove.approved
  if: "author_association == 'CONTRIBUTOR'"
  comment: "The review is completed. Thanks @{{ author }}, we'll take it from here."

groups:
  test-core:
    reviewers:
      teams:
      - test-core
    reviews:
      required: 1  # number of approvals required from this group
      request: 1  # request from all members a review
      request_order: shuffle
```

**Comment by adrianjost** [comment]
/pullapprove-test
```
version: 3
groups:
  config-test:
    config-test:
      conditions:
      - "'wontfix' in labels"
      reviewers:
        users: ['WorldofJARcraft']
```

File: .pullapprove.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  S⊙w⊙T:
+    conditions:
+    - "'Team S⊙w⊙T' in labels"
+    reviewers:
+      teams:
+      - s-w-t
+    reviews:
+      required: 1
+      request: 1
+  TEAM-AB:
```
Comment (adrianjost) [inline .pullapprove.yml]:
```suggestion
  RESTless:
```

File: .pullapprove.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    conditions:
+    - "'Team AB' in labels"
```
Comment (adrianjost) [inline .pullapprove.yml]:
```suggestion
    - "'Team RESTless' in labels"
```

**Pull request merged**